### DISCUSSION TITLE
Still tied to Mapbox?

### DISCUSSION THREAD

**Discussion by scenicfinder** [maplibre/maplibre-navigation-android#101] [Q&A]
Hi, I have my dev trying to implement maplibre navigation into our app, however he says it is using mapbox access tokens in some places? We were initially using mapbox, and he was having no trouble with it, but now we are kind of stuck changing over to an alternative.

We also want to have POI icons, routing that follows pre-determined routes and some other features. For map hosting we want to use either protomaps or stadiamaps. I do know protomaps uses the MVT format.

So whilst mapbox is simply not viable financially for us, it seems the alternatives are a lot more complex. We are kind of stuck...

My dev says: (please excuse his English)

> "In the github source code of Maplibre, it's still using mapbox some place (check my screenshot) in case you want to change the maphost, but there is no document to teach how to change from mapbox to another host, we will not know how to update the code above. At least the github source of the navigation must design as a plus and we can plug-in to any sockets (map host), should have a document to teach this work."

![image](https://github.com/maplibre/maplibre-navigation-android/assets/156398576/a349eba7-cf5e-42a9-b856-977322c6d148)

**Comment by boldtrn** [comment]
The default UI (navigation-ui) is still using the MapboxDirections SDK. You can use this SDK with other services than Mapbox though, you could use it with GraphHopper or Valhalla for example. You can use this repo without signing up for a Mapbox API key. If you want to, you can use it with Mapbox though.

> For map hosting we want to use either protomaps or stadiamaps. I do know protomaps uses the MVT format.

While not related to navigation, I think both should work fine with Maplibre-Native and this repo.

With that said, if you need some features from this repo or if something is not working, feel free to let us know and ideally create a PR to fix it 👍

**Reply by JonasVautherin** [reply]
As mentioned in https://github.com/maplibre/maplibre-navigation-android/issues/16#issuecomment-1912440312, I still get `com.mapbox.core.exceptions.ServicesException: Using Mapbox Services requires setting a valid access token` when running both the [maplibre example](https://github.com/maplibre/maplibre-navigation-android) and the [graphhopper example](https://github.com/graphhopper/graphhopper-navigation-example).

I understand that it *should* not require Mapbox, but it seems like it does for me. What could I be missing? :innocent:

**Comment by Mario1R** [comment]
Any news on this? I tried to juggle with dependencies and what not for a week now... I read that the UI library was excluded from the SDK (why?) Is there a different function, a set of libraries/versions that I am supposed to call/implement instead? Since there is little documentation I'm struggling to understand this SDK. (maplibre's navigation SDK version is 3.0.0 in this case, however I tried 2.0.0 just to see if it makes any difference)
![help](https://github.com/user-attachments/assets/696d5045-203c-4862-b587-5619db8157be)
![what](https://github.com/user-attachments/assets/5b7f98ec-be26-4fbb-b5b3-70250b684641)
